obspy/core/inventory/network.py

```diff
@@ -275,16 +275,55 @@ def get_coordinates(self, seed_id, datetime=None):
                     # if channel latitude or longitude is not given use station
                     data['latitude'] = cha.latitude or sta.latitude
                     data['longitude'] = cha.longitude or sta.longitude
-                    data['elevation'] = cha.elevation
+                    data['elevation'] = cha.elevation or sta.elevation
```

Comment:
Minor nitpick as its a bit unlikely in practice - but the three previous lines should be `cha.X if cha.X is not None else sta.X`.